File: taxifleet_backend/src/main/java/ru/taxifleet/entity/Car.java
package ru.taxifleet.entity;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import ru.taxifleet.enums.CarStatus;

@Entity
@Table(name = "cars")
@Getter @Setter @NoArgsConstructor
@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
public class Car {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "license_plate", nullable = false, unique = true, length = 20)
    private String licensePlate;

    @Column(nullable = false, length = 50)
    private String brand;

    @Column(nullable = false, length = 50)
    private String model;

    @Column(nullable = false)
    private Integer year;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private CarStatus status = CarStatus.AVAILABLE;

    @Column(name = "mileage_km", nullable = false)
    private Integer mileageKm = 0;

    public boolean isAvailable() {
        return this.status == CarStatus.AVAILABLE;
    }

    public void sendToMaintenance() {
        this.status = CarStatus.MAINTENANCE;
    }
}

File: taxifleet_app/lib/models/car.dart
class Car {
  final int id;
  final String licensePlate;
  final String brand;
  final String model;
  final int year;
  final String status;
  final double mileageKm;

  Car({
    required this.id,
    required this.licensePlate,
    required this.brand,
    required this.model,
    required this.year,
    required this.status,
    required this.mileageKm,
  });

  factory Car.fromJson(Map<String, dynamic> json) {
    return Car(
      id: json['id'] as int,
      licensePlate: json['licensePlate'] as String? ?? '',
      brand: json['brand'] as String? ?? '',
      model: json['model'] as String? ?? '',
      year: json['year'] as int? ?? 0,
      status: json['status'] as String? ?? 'UNAVAILABLE',
      mileageKm: (json['mileageKm'] as num?)?.toDouble() ?? 0.0,
    );
  }
}
